### ISSUE TITLE
Download from iOS don't run

### ISSUE THREAD

**Issue by amedeopro** [fluttercommunity/flutter_downloader#624]
Hello,
I am trying to download mp3 files from a playlist, on android no problem but on iOS the download does not seem to start but it does not give any error or warning it is simply as if it ignored the download function

```
Future<void> downloadFile(fileName, md5) async {
    final externalDir = Platform.isAndroid ? await getExternalStorageDirectory() : await getApplicationDocumentsDirectory();
      FlutterDownloader.enqueue(
          url: '${dotenv.env['API_FILE_DOWNLOAD']}${fileName}',
          savedDir: externalDir!.path,
          fileName: "${md5}.mp3",
          showNotification: true,
          openFileFromNotification: true,
      );
  }
  
```

Can you help me?
Thank you

**Comment by adminant** [comment]
I have the same problem. Download works ok on Android, but on iOS DownloadTaskStatus after 0 becomes 4 and nothing is downloaded.

**Comment by arunraj-qburst** [comment]
I am facing same issue (iOS only) .

Here is the log which I can see when I call the open file,

```
Runner[1637:321675] execute query: SELECT * FROM task WHERE task_id = "<my-app-id>.download.background.1649618761.683855.3" ORDER BY id DESC LIMIT 1
Runner[1637:321675] open DB successfully
Runner[1637:321675] Load task successfully
Runner[1637:323574] DB Error: database is locked
Runner[1637:323574] Query was executed successfully. Affected rows = 1
Runner[1637:323574] URLSession:task:didCompleteWithError:
Runner[1637:323574] HTTP status code: 200
Runner[1637:321906] flutter: not found task corresponding to given task id
```

Flutter version 2.10.4
flutter_downloader: ^1.7.1
iOS ver 14+

**Comment by AyumiSashitani** [comment]
Hi. everyone,
I faced same issue.

And I solved by below link
https://github.com/fluttercommunity/flutter_downloader/issues/611#issuecomment-1059124963

Thank you.

**Comment by atleugim** [comment]
> Hi. everyone, I faced same issue.
> 
> And I solved by below link [#611 (comment)](https://github.com/fluttercommunity/flutter_downloader/issues/611#issuecomment-1059124963)
> 
> and
> 
> https://stackoverflow.com/a/66380363/17725279
> 
> Thank you.

Thanks, this solved mi issue with downloading and opening files, but where is the file? i don't see it in the Files app anywhere

**Comment by bartekpacia** [comment]
Assuming this is resolved.

If not, please create a new bug report using the new [issue template](https://github.com/fluttercommunity/flutter_downloader/issues/new/choose).
